src/deque.cr

```diff
@@ -581,15 +581,39 @@ class Deque(T)
     end
   end
 
-  private def increase_capacity
+  private INITIAL_CAPACITY = 4
+
+  # behaves like `calculate_new_capacity(@capacity + 1)`
+  private def calculate_new_capacity
+    return INITIAL_CAPACITY if @capacity == 0
+
+    @capacity * 2
+  end
```

Comment:
This overload is never called. Does it still make sense to add it?

Reply:
There were originally separate overloads for `#resize_if_cant_insert` too:

```crystal
private def resize_if_cant_insert
  new_capacity = calculate_new_capacity
  if new_capacity > @capacity
    resize_to_capacity(new_capacity)
  end
end

private def resize_if_cant_insert(insert_size)
  new_capacity = calculate_new_capacity(@size + insert_size)
  if new_capacity > @capacity
    resize_to_capacity(new_capacity)
  end
end
```

Then `#calculate_new_capacity` would be used here whereas `#calculate_new_capacity(new_size)` would remain unused until #13283. This would get rid of the `while` for methods other than `#concat`. What do you think?

Reply:
Maybe just add it in when it's actually used in https://github.com/crystal-lang/crystal/pull/13283?